File: cpp/Easy/70_climbing_stairs.cpp
class Solution {
public:
int climbStairs(int n) {
    int ans;
    if(n == 0) return 0;
    if(n == 1) return 1;
    int prev = 1, curr = 1;
    for(int i = 2; i<=n; i++){
        ans = prev + curr;
        prev = curr;
        curr = ans;
    }
    return ans;
    }
};

File: python/easy/70_climbing_stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        deck = {}
        deck[1] = 1
        deck[2] = 2
        def sub (n, deck):
            if n in deck:
                return deck[n]
            else:
                deck[n] = sub(n-1, deck) + sub(n-2, deck)
                return deck[n]
        return sub(n, deck)
